src/nomad_tajine_plugin/schema_packages/schema_package.py

```diff
@@ -103,24 +102,60 @@ class Ingredient(EntityReference):
     # preparation_notes = Quantity() or SubSection() TODO: discuss
     # TODO: discuss references
 
-    def normalize(self, archive, logger: 'BoundLogger'):
+    def normalize(self, archive, logger: 'BoundLogger'):  # noqa: PLR0912
         if not self.lab_id:
             self.lab_id = self.name
 
         super().normalize(archive, logger)
 
-        # if self.reference:
-        #     match self.unit:
-        #         case 'gram':
-        #             self.quantity_si = self.quantity
-        #         case 'piece':
-        #             self.quantity_si = self.reference.weight_per_piece * self.quantity
-        #         case _:
-        #             self.quantity_si = (
-        #                 (ureg(self.unit).to(ureg.milliliter)).magnitude
-        #                 * self.quantity
-        #                 * self.reference.density
-        #             )
+        if self.reference:
+            unit = self.unit.replace(' ', '_')  # type: ignore
+            match unit:
+                case 'gram':
+                    self.quantity_si = self.quantity
+                case 'piece':
+                    if self.reference.weight_per_piece:
+                        self.quantity_si = (
+                            self.reference.weight_per_piece * self.quantity
+                        )
+                    else:
+                        self.quantity_si = None
+                case 'teaspoon':
+                    if self.reference.density:
+                        self.quantity_si = (
+                            ureg.Quantity(14.79, 'milliliter')
+                            * self.quantity
+                            * self.reference.density
```

Comment:
Quantity `density` is typically reported as kg/m^3 or g/cm^3, using kg/l can take advantage of that 1 kg/l is 1 g/ml so jep